src/shared/components/palette/address/address-option.model.ts

```diff
@@ -1,21 +1,34 @@
 import { AddressModel } from '../../../domain/addresses/address.model';
 export class AddressOption {
 
-  value: AddressModel;
   description: string;
 
+  value: AddressModel;
+
   constructor (addressModel: AddressModel, description: string) {
-    this.value = addressModel;
-    this.description = (description == null)
-        ? this.value.AddressLine1
-            +  this.prefixWithCommaIfPresent(this.value.AddressLine2)
-              +  this.prefixWithCommaIfPresent(this.value.AddressLine3)
-                + ', ' + this.value.PostTown
-                  : description;
+    if(description == null) {
+      this.value = addressModel;
+      this.description = this.getDescription();
+    } else {
+      this.description = description;
```

Comment:
are we forgetting to set the 
`this.value = addressModel;`
in this branch?

Reply:
when the description is passed through constructor it is only used to hold default description that has a total address label on drop down field, We can set it but will never be used.
-----------------------------
 total 12 address found            |   <--------- default option value (show the total count )
-----------------------------
1, Bond street, London W15L
2, Bond Street, London W15L